File: internal/content/testdata/bench/go/nc-hand-of-straights.go
import "sort"

func is_n_straight_hand(hand []int, group_size int) bool {
	if len(hand)%group_size != 0 {
		return false
	}
	count := make(map[int]int)
	for _, c := range hand {
		count[c]++
	}
	keys := make([]int, 0, len(count))
	for k := range count {
		keys = append(keys, k)
	}
	sort.Ints(keys)
	for _, card := range keys {
		if count[card] > 0 {
			need := count[card]
			for i := 0; i < group_size; i++ {
				if count[card+i] < need {
					return false
				}
				count[card+i] -= need
			}
		}
	}
	return true
}

File: internal/content/testdata/bench/csharp/nc-hand-of-straights.cs
using System.Collections.Generic;

public class Solution {
    public bool is_n_straight_hand(long[] hand, long group_size) {
        if (hand.Length % group_size != 0) return false;
        var count = new SortedDictionary<long, long>();
        foreach (long card in hand) {
            if (!count.ContainsKey(card)) count[card] = 0;
            count[card]++;
        }
        foreach (long card in new List<long>(count.Keys)) {
            long need = count.ContainsKey(card) ? count[card] : 0;
            if (need > 0) {
                for (long i = 0; i < group_size; i++) {
                    long c = count.ContainsKey(card + i) ? count[card + i] : 0;
                    if (c < need) return false;
                    count[card + i] = c - need;
                }
            }
        }
        return true;
    }
}
